src/main/scala/io/gatling/build/sonatype/GatlingSonatypePlugin.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2011-2024 GatlingCorp (https://gatling.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.gatling.build.sonatype
+
+import io.gatling.build.basic.GatlingBasicInfoPlugin
+import io.gatling.build.basic.GatlingBasicInfoPlugin.GatlingBasicInfoKeys._
+import io.gatling.build.publish.GatlingPublishPlugin
+import io.gatling.build.release.GatlingReleasePlugin
+import io.gatling.build.release.GatlingReleasePlugin.GatlingReleaseKeys._
+
+import com.jsuereth.sbtpgp.PgpKeys.publishSigned
+import sbtrelease.ReleasePlugin.autoImport._
+import sbtrelease.ReleaseStateTransformations.reapply
+import xerial.sbt.Sonatype
+import xerial.sbt.Sonatype.SonatypeKeys._
+
+import sbt._
+import sbt.Keys._
+
+object GatlingSonatypePlugin extends AutoPlugin {
+  override def requires: Plugins =
+    GatlingBasicInfoPlugin &&
+      GatlingReleasePlugin &&
+      GatlingPublishPlugin &&
+      Sonatype
+
+  trait GatlingSonatypeKeys {
+    val gatlingPublishToSonatype = settingKey[Boolean]("true to publish to sonatype")
+  }
+  object GatlingSonatypeKeys extends GatlingSonatypeKeys
```

Comment:
redundant name?
Either as an first citizen, or rename it
```suggestion
  trait Keys {
    val gatlingPublishToSonatype = settingKey[Boolean]("true to publish to sonatype")
  }
  object Keys extends Keys
```

Reply:
This was to follow the same convention as the other existing plugins (e.g. `GatlingBasicInfoPlugin.GatlingBasicInfoKeys`, `GatlingPublishPlugin.GatlingPublishKeys`, etc.)

Reply:
Oh! Ok :disappointed: 

Maybe we should have our own convention :shrug: 

But let's keep it for now